forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -618,6 +618,15 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         // Need to apply any static effects to produce correct triggers
         checkStaticAbilities();
 
+        // 400.7g try adding keyword back into card if it doesn't already have it
+        if (zoneTo.is(ZoneType.Stack) && cause != null && cause.isSpell() && c.equals(cause.getHostCard())) {
+            if (cause.getKeyword() != null) {
+                if (!copied.getKeywords().contains(cause.getKeyword())) {
```

Comment:
the contains check is for full equalness, not sure if that was intended

Currently if I grant it a second flashback via Snapcaster and cast it for the intrinsic both will want to exile it:
![grafik](https://user-images.githubusercontent.com/8506892/217750908-0b56d89f-e62e-42f5-971c-ba47fd99ac76.png)

I'm not sure if that's correct

Reply:
it shouldn't be a problem because both of them does exile, but we might do "castKeyword" onto it like done with Dash

Reply:
the full equalness was on purpose so i can check if the Card on the Stack doesn't has the same keyword instance as i want to add.

Reply:
@tool4ever 
can you check if `+castKeyword` works in this one?
https://github.com/Card-Forge/forge/blob/b4e95b47e398c762119c1c8ea980237237eac509/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java#L2287-L2288

if it works for ValidCard or ValidStackSa ?

Reply:
Doesn't work

`getKeywordForStaticAbility` always calls `createTraits` with intrinsic=false

Seems wrong too?

Reply:
> Doesn't work
> 
> `getKeywordForStaticAbility` always calls `createTraits` with intrinsic=false
> 
> Seems wrong too?

which one? 

it shouldn't be a problem i think, otherwise we might need to refactor the "castKeyword" property

or where does it fail?

Reply:
it breaks all flashback exiling

the check in the property fails because it's not seen as intrinsic

Reply:
what happens if we remove the intristic check and let the Keyword just compare with each other?

or does this break with cards with intrinsic Flashback?

Reply:
That seems ok for flashback, but I'm worried Henzie needed it?

Also isn't the problem that the static of Virals own flashback should be intrinsic too?

Reply:
For Henzie it should work because it goes over getKeywordForStaticAbility, so it should be the same object?

You need to try it with Henzie vs a Card with Natural Blitz if both still works

For Flashback it should probably set the copied Spell to intrinsic if the Keyword is intrinsic, Virals itself isn't the problem i think

Reply:
right now it looks like the flashback copy just keeps the original intrinsic while the RE will always be non-intrinsic

Will test with Blitz in a bit

Reply:
what happens if you would remove this check?
https://github.com/Card-Forge/forge/blob/b4e95b47e398c762119c1c8ea980237237eac509/forge-game/src/main/java/forge/game/card/CardProperty.java#L1825-L1830

does it still work for Henzie and normal Blitz?

Reply:
No, it would stop Haste of intrinsic Blitz :/

Reply:
i think i'm in some kind of dilemma:
* For Flashback: Keyword currently don't know if they are intrinsic after they were created, so i might not set the correct value to the copied Flashback Spell
* for Intrinsic Blitz, The Keyword which used to cast the Creature with might not be the same keyword it has on the battlefield. (because Card object gets copied?)

but i will think about it

Reply:
I'm was surprised it didn't create a secondary Keyword for intrinsic Blitz
but then i looked an it is created differently